src/Functions/seriesDecomposeSTL.cpp

```diff
@@ -94,10 +94,19 @@ class FunctionSeriesDecomposeSTL : public IFunction
                 || checkAndGetColumn<ColumnUInt32>(period_ptr.get())
                 || checkAndGetColumn<ColumnUInt64>(period_ptr.get()))
                 period = period_ptr->getUInt(i);
+            else if (checkAndGetColumn<ColumnFloat32>(period_ptr.get())
+                || checkAndGetColumn<ColumnFloat64>(period_ptr.get()))
+            {
+                auto period_val = period_ptr->getFloat64(i);
+                if (isnan(period_val) || !std::isfinite(period_val) || period_val < 0.0)
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "The second argument for function {} should be a positive number", getName());
```

Comment:
It said that period should be positive but this condition accepts `period_val = 0`

Reply:
also ref l.109